hoprd/rest-api/src/preconditions.rs

```diff
@@ -61,3 +62,16 @@ pub(crate) async fn authenticate(
     // Go forward to the next middleware or request handler
     next.run(request).await
 }
+
+pub(crate) async fn ensure_running(
+    State(state): State<InternalState>,
+    _uri: OriginalUri,
+    _headers: HeaderMap,
+    request: Request,
+    next: Next,
+) -> impl IntoResponse {
+    if state.hopr.status() != HoprState::Running {
+        return (StatusCode::PRECONDITION_FAILED, "Node is not ready to accept requests").into_response();
+    }
+    next.run(request).await
```

Comment:
The biggest beef I'd have with  this setup, is that this check happens at least twice at this point. Once in the rest API, once in the `hopr_lib::Hopr` itself. If all the API functionality needs to know this precondition, should it rather not be all done on the `hopr_lib::Hopr` level? The answer likely is yes, because neither the API can be polled, when the node is not ready. But then there is some functionality, which can be polled, so the decision should lie in the hopr_lib itself.

Reply:
All the API endpoints doesn't need this precondition, it's only applied to a subset of them.

Reply:
But that's the point. All API endpoints are using the `hopr_lib::Hopr` object. If there's a call that needs to check whether the node is in a usable state, this check should already be in the `hopr_lib::Hopr`. And indeed that's precisely what is happening: https://github.com/hoprnet/hoprnet/blob/031b7bbcb8d93ba1d3b9e0402ee23242021f7245/hopr/hopr-lib/src/lib.rs#L1207

So what needs to happen if you want to designate the a special 412 error is to pattern match against the error returned by that API.

Reply:
The REST API structure does not need to be touched at all.

Reply:
https://github.com/hoprnet/hoprnet/blob/master/hopr/hopr-lib/src/lib.rs#L570-576